### DISCUSSION TITLE
sonoff miniR2 not working anymore

### DISCUSSION THREAD

**Discussion by Furtivo360** [RavenSystem/esp-homekit-devices#2017] [MEPLHAA Scripts]
**Detailed steps to reproduce it**
After latest update to Merlin 12.0.3, my devices are not working anymore.
I can see them connected from the HAA app, I can access the setup page, but cannot turn ON/OFF lights anymore. Nor from app nor from physical switch.

**MEPLHAA Script**

{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0},{"g":4},{"g":4,"t":0}]}]}

**Used device**

Sonoff miniR2

**HAA version**

Merlin 12.0.3

Please help asap…I cannot turn lights on or off…

thanks

**Comment by xrust83** [comment]
no offense, but if you have a problem try to read the problems of other users before opening the same problem 100 times, the last 2 weeks in each topic there is an explanation of your problem

https://github.com/RavenSystem/esp-homekit-devices/wiki/Migration-Guide-for-HAA-V12-Merlin

**Comment by xrust83** [comment]
{"c":{"io":[[[13],3],[[0,4],6],[[12],2]],"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0},{"g":4},{"g":4,"t":0}]}]}


For all Sonoff mini "io":[[[13],3],[[0,4],6],[[12],2]],

For all Sonoff basic "io":[[[13],3],[[0],6],[[12],2]],

**Reply by tvatheuer** [reply]
It is working with a switch on external GPIO but not with a button. I changed it to work with a button as a Toggle button for switching :
{"c":{"io":[[[13],3],[[0,4],6],[[12],2]],"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0},{"g":4}]}]}
